src/ResourceManagement/Network/Microsoft.Azure.Management.Network.Fluent/ApplicationGatewayImpl.cs

```diff
@@ -0,0 +1,1496 @@
+        }
+
+        ///GENMHASH:EE79C3B68C4C6A99234BB004EDCAD67A:75C2DBDE4D61EDB193493A39766B60EE
+        public ApplicationGatewayImpl WithExistingSubnet(ISubnet subnet)
+        {
+            EnsureDefaultIpConfig().WithExistingSubnet(subnet);
+            return this;
+        }
+
+        ///GENMHASH:5647899224D30C7B5E1FDCD2D9AAB1DB:E44EB8F969AE406A3CE854143982FB65
+        public ApplicationGatewayImpl WithExistingSubnet(INetwork network, string subnetName)
+        {
+            EnsureDefaultIpConfig().WithExistingSubnet(network, subnetName);
+            return this;
+        }
+
+        ///GENMHASH:E8683B20FED733D23930E96CCD1EB0A2:7246BAAE40C91731DF4E96029B3FA2BA
+        public ApplicationGatewayImpl WithExistingSubnet(string networkResourceId, string subnetName)
+        {
+            EnsureDefaultIpConfig().WithExistingSubnet(networkResourceId, subnetName);
+            return this;
+        }
+
+        ///GENMHASH:E38530BE2EC4569C8D62DB8CEB4AD38F:90CC3E1EEF35BDBB9152B8B5149B04E0
+        private ApplicationGatewayIpConfigurationImpl DefineIpConfiguration(string name)
+        {
+            IApplicationGatewayIpConfiguration config = null;
+            if (!ipConfigs.TryGetValue(name, out config))
+            {
+                var inner = new ApplicationGatewayIPConfigurationInner()
+                {
+                    Name = name
+                };
+                return new ApplicationGatewayIpConfigurationImpl(inner, this);
+            }
+            else
+            {
+                return (ApplicationGatewayIpConfigurationImpl)config;
+            }
+        }
+
+        ///GENMHASH:A3E0AFFD41A48AADA625D444BDC4B639:D755BC44A5AE232FE3D5AB7294B7260E
+        public ApplicationGatewayImpl WithoutIpConfiguration(string ipConfigurationName)
+        {
+            ipConfigs.Remove(ipConfigurationName);
+            return this;
+        }
+
+        #endregion
+
+        #region Listeners
+
+        ///GENMHASH:335DEBA2C3ED42B7D6D726224668713C:5F7D61C72418DAEB7F935725E2A62AFF
+        public ApplicationGatewayListenerImpl DefineListener(string name)
+        {
+            IApplicationGatewayListener httpListener = null;
+            if (!listeners.TryGetValue(name, out httpListener))
+            {
+                var inner = new ApplicationGatewayHttpListenerInner()
+                {
+                    Name = name
+                };
+                return new ApplicationGatewayListenerImpl(inner, this);
+            }
+            else
+            {
+                return (ApplicationGatewayListenerImpl)httpListener;
+            }
+        }
+
+        ///GENMHASH:FD3BFA79E44BF0C0FD92A0CE7B31B143:4D125EAA0586FE83630989875156AFFD
+        internal ApplicationGatewayImpl WithHttpListener(ApplicationGatewayListenerImpl httpListener)
+        {
+            if (httpListener == null)
+            {
+                return null;
```

Comment:
ok, so since this is an internal function, not user facing, and there isn't really a user scenario where this would get null as the input, I'm simplifying this logic now to not return null at all, just ignore if null is passed.